src/cemerick/friend/util.clj

```diff
@@ -12,9 +12,14 @@
 ; this was never _really_ part of the API, was implemented (badly) before req/request-url was available
 (def ^:deprecated original-url req/request-url)
 
-(defn resolve-absolute-uri
-  [^String uri request]
-  (-> (original-url request)
-    java.net.URI.
-    (.resolve uri)
-    str))
+
+(defn relative-url
+  "Similar to `util/original-url` (and thus `ring.util.request/request-url`)
+  except that it only uses `:uri` and `:query-string` parts of the request
+  to build the URL and ignores `:scheme` and the 'host' header.
+  This is to avoid issues with redirects using absolute URLs when the app is running behind an SSL proxy."
+  [request]
+  (str
+   (:uri request)
+   (when-let [query (:query-string request)]
+     (str "?" query))))
```

Comment:
Compare to [`ring.util.request/request-url`](https://github.com/ring-clojure/ring/blob/master/ring-core/src/ring/util/request.clj#L9-L14).